core/extension/gdextension_interface.h

```diff
@@ -1526,6 +1526,19 @@ typedef void (*GDExtensionInterfaceStringOperatorPlusEqWcstr)(GDExtensionStringP
  */
 typedef void (*GDExtensionInterfaceStringOperatorPlusEqC32str)(GDExtensionStringPtr p_self, const char32_t *p_b);
 
+/**
+ * @name string_resize
+ * @since 4.2
+ *
+ * Resizes the underlying string data to the given number of characters.
```

Comment:
Might want to specify that the string needs to be null terminated manually

Reply:
Ah, yeah, that's a good idea! I've added a note about the null terminating character in my latest push